tests/AsyncBridge.Tests/CancelAfterTests.cs

```diff
@@ -69,14 +70,76 @@ public void CancelRemove()
         {
             TestUtils.RunAsync(async () =>
             {
-                using (var Source = new CancellationTokenSource())
+                using (var cancelSource = new CancellationTokenSource())
                 {
-                    Source.CancelAfter(50);
-                    Source.CancelAfter(Timeout.Infinite);
+                    cancelSource.CancelAfter(50);
+                    cancelSource.CancelAfter(Timeout.Infinite);
 
                     await TaskEx.Delay(250);
 
-                    Assert.IsFalse(Source.IsCancellationRequested);
+                    Assert.IsFalse(cancelSource.IsCancellationRequested);
+                }
+            });
+        }
+
+        [TestMethod]
+        public void CancelConcurrentNew()
+        {
+            TestUtils.RunAsync(async () =>
+            {
+                using (var cancelSource = new CancellationTokenSource())
+                {
+                    var continueSource = new TaskCompletionSource<bool>();
+
+                    // Spawn ten threads to create an immediate timer
```

Comment:
Queuing ten work items on the thread pool doesn't actually spawn ten threads, as I found out earlier today when reporting a race condition in TypeDescriptor.

Reply:
I ended up using a CountdownEvent-like object (Interlocked.Decrement + MRES), but then happily I didn't actually end up needing such precision and deleted the code.

Reply:
As long as the test seems to both fail and pass as expected, all I really care about is making sure the comment is explaining it well.